### PR TITLE
Feature/redis cache try get async

### PR DESCRIPTION
**Pull request by gregoks**

### CODE REVIEW

File: RedisCaching/CacheContext.cs [hunk 1]
```diff
@@ -40,13 +40,13 @@ internal void SetItem<TValue>(CacheItem<TValue> cacheItem)
             client.Database.StringSet(cacheItem.Id, convert.ToRedis(cacheItem), expiry, When.Always, CommandFlags.FireAndForget);
         }
 
-        internal Task SetItemAsync<TValue>(CacheItem<TValue> cacheItem)
+        internal async Task SetItemAsync<TValue>(CacheItem<TValue> cacheItem)
```
Comment (amitla) [inline RedisCaching/CacheContext.cs]: @gregoks I don't see reason to make this function async (it was already awaitable).

File: RedisCaching/CacheContext.cs [hunk 2]
```diff
@@ -40,13 +40,13 @@ internal void SetItem<TValue>(CacheItem<TValue> cacheItem)
...
         {
             TimeSpan? expiry = null;
             if (cacheItem.ExpireIn.HasValue)
                 expiry = cacheItem.ExpireIn;
 
-            return client.Database
+            await client.Database
                 .StringSetAsync(cacheItem.Id, convert.ToRedis(cacheItem), expiry, When.Always,
                     CommandFlags.FireAndForget);
```
Comment (amitla) [inline RedisCaching/CacheContext.cs]: if **there is** a justification to make it async, you have to add 
```suggestion
       CommandFlags.FireAndForget).ConfigureAwait(false);
```
to your await line.

**Comment by amitla** [comment]
@gregoks I've added a couple of comments in CacheContext.cs but then I noticed that my comments apply to all of your changes in that file.

Two things:

1. A method that returns a Task is already awaitable, you should only make it async to **await** inside. Do this only if there is good justification.
2. If you do need a method to be async, you should always await with ``` .ConfigureAwait(false) ```

File: RedisCaching/CacheContext.cs (hunk 1 above)
Comment (gregoks) [reply RedisCaching/CacheContext.cs]: Added ConfigureAwait since we need it to be awaitable all the way.

**Pull request merged**